build.gradle

```diff
@@ -894,4 +894,15 @@ publishing {
             }
         }
     }
-}
\ No newline at end of file
+}
+
+// updateVersion: Task to auto increment to the next development iteration
+task updateVersion {
+    onlyIf { System.getProperty('newVersion') }
+    doLast {
+        ext.newVersion = System.getProperty('newVersion')
+        println "Setting version to ${newVersion}."
+        // String tokenization to support -SNAPSHOT
+        ant.replaceregexp(file:'build.gradle', match: '"opensearch.version", "\\d.*"', replace: '"opensearch.version", "' + newVersion.tokenize('-')[0] + '-SNAPSHOT"', flags:'g', byline:true)
```

Comment:
We also have BWC tests where we test the upgrade with previous major version(i.e. `opensearch_previous_version`). 

This method should also upgrade the `opensearch_previous_version` if there is a major version bump for `opensearch.version`.

Reply:
Hey @ankitkala  Once the cross-cluster-replication is aligned to [core repo branching ](https://github.com/opensearch-project/opensearch-plugins/issues/142) (Related [issue](https://github.com/opensearch-project/cross-cluster-replication/issues/443)), since `main` will be the next major version, so do you think it has to be part of `version increment`?
@bbarani

Reply:
@prudhvigodithi Regarding the migration to core, we've are yet to figure out how to completely move to core. Most likely, we'll still require a separate plugin (to start with) and eventually would move completely to core. 

But anyways, the BWC tests are not specific to CCR plugin. Don't have to necessarily do in this PR but still we should sort that part out. What do you think?

Reply:
Hey @ankitkala are we on the same page? :), I mean CCR to align with core branching strategy, CCR can still be a separate plugin but uses same as core branching strategy, `main` will be the next major version, `<major version>.x` (example `2.x`) will be next minor version and 2.1 or 2.2 will be for hotfix/patch versions.
If we follow this way `opensearch_previous_version` can be maintained as per branch for BWC tests alone.

Reply:
My bad. Please ignore my comment around "move to core".

I got what you mean with the branching strategy. Basically we only modify the `opensearch_previous_version` when we move to a new major version. 
But if we're  planning to use this automation for major version bump as well, then we should be handling this IMHO. Otherwise its just one more thing plugin owners need to remember.  As an alternative, if you want to keep that as a checklist item in every new major release, that is fine as well.
Thoughts?

Reply:
Added a tracking issue for this: https://github.com/opensearch-project/opensearch-build/issues/2373